iotdb/src/main/java/org/apache/iotdb/db/query/control/QueryResourceManager.java

```diff
@@ -34,13 +37,17 @@
 
 /**
  * <p>
- * Singleton pattern, to manage all query tokens. Each jdbc query request can query multiple series,
- * in the processing of querying different device id, the <code>FileNodeManager.getInstance().
- * beginQuery</code> and <code>FileNodeManager.getInstance().endQueryForGivenJob</code> must be invoked in the
- * beginning and ending of jdbc request.
+ * QueryResourceManager manages resource (file streams) used by each query job, and assign Ids to the jobs.
+ * During the life cycle of a query, the following methods must be called in strict order:
+ * 1. assignJobId - get an Id for the new job.
+ * 2. beginQueryOfGivenQueryPaths - remind FileNodeManager that some files are being used
+ * 3. (if using filter)beginQueryOfGivenExpression

```

Comment:
if using filter, both step 2 and 3 need to be called? or just either of them?

Reply:
Both, as you can see, there is no condition on the second step.